service/tablet_allocator.hh

```diff
@@ -129,13 +129,13 @@ public:
 // It's an average per-shard load in terms of tablet count.
 using load_type = double;
 
-enum class tablet_replica_calculation_status {
-    success,
-    no_tablets_to_allocate,
-    not_enough_nodes,
-    more_than_one_error,
-    unknown_topology_strategy,
+enum tablet_replica_calculation_status {
+    success = 1,
+    not_enough_nodes = 2,
+    unknown_topology_strategy = 3,
+    other = 4,
```

Comment:
why the numerical enumeration and how is it related to this patch?
if the intention is to reserve some unused value slots for backward compatibility, better rename them to e.g. `_unused_no_tablets_to_allocate`.

Reply:
This enum is introduced in the very 1st commit of this patch series (authored by @KrzaQ ). I am serializing the status in 1 place to a numeric code and thought this debugging info could prove useful.

Reply:
I'm not sold on the need for returning tablet_replica_calculation_status at all.
It seems to complicate the code and error handling for no good reason.

Cc @KrzaQ

Reply:
I removed this enum altogether.